src/web/routes/application/flow-control/middleware/handle-path-request/handle-path-request.js

```diff
@@ -1,18 +1,19 @@
 const { CONFIRM_URL } = require('../../../paths')
 const { stateMachine, actions, states } = require('../../state-machine')
-const { isPathInApplicationFlow, stepNotNavigable } = require('./predicates')
-const { getPathsInSequence } = require('./selectors')
+const { stepNotNavigable } = require('./predicates')
 
 const { COMPLETED } = states
 const { IS_PATH_ALLOWED, GET_NEXT_ALLOWED_PATH, SET_NEXT_ALLOWED_PATH } = actions
 
-const middleware = (config, pathsInSequence, step) => (req, res, next) => {
+const handleRequestForPath = (config, journey, step) => (req, res, next) => {
+  const { pathsInSequence } = journey
+
   // Destroy the session on navigating away from CONFIRM_URL
   if (stateMachine.getState(req) === COMPLETED && req.path !== CONFIRM_URL) {
     req.session.destroy()
     res.clearCookie('lang')
 
-    if (isPathInApplicationFlow(req.path, pathsInSequence)) {
+    if (pathsInSequence.includes(req.path)) {
```

Comment:
This seemed like an unnecessary abstraction as the unit tests were testing core javascript. I think the inline code reads better than the abstraction.